app/main.py

```diff
@@ -1,8 +1,44 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list) -> float:
+        income = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += self.calculate_washing_price(car)
+            self.wash_single_car(car)
+        return round(income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        clean_dif = self.clean_power - car.clean_mark
+        pr_without_dist = car.comfort_class * clean_dif * self.average_rating
+        price = pr_without_dist / self.distance_from_city_center
+        return round(price, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, single_rate: int) -> None:
+        sum_rate = (self.average_rating * self.count_of_ratings) + single_rate
+        new_rate = round((sum_rate / (self.count_of_ratings + 1)), 1)
+        self.average_rating = new_rate
+        self.count_of_ratings = (self.count_of_ratings + 1)
```

Comment:
Clean Code Issue: The calculation of the new service rating in the rate_service method is a bit complex. It would be better to split it into several steps with intermediate variables to make it more readable. Also, you can avoid using parentheses in the last line of this method. There's no need to use parentheses when you're just adding 1 to a variable.